deeprank_gnn/NeuralNet.py

```diff
@@ -293,39 +342,41 @@ def train(self, nepoch=1, validate=False, save_model='last',
 
                 train_losses.append(loss_)
 
-
                 # Validate the model
                 if validate:
 
-                    t0 = time()
+                    # t0 = time()
                     loss_ = self.eval(self.valid_loader, epoch, "validation")
-                    t = time() - t0
+                    # t = time() - t0
 
                     valid_losses.append(loss_)
 
-                    # save the best model (i.e. lowest loss value on validation data)
+                    # save the best model (i.e. lowest loss value on validation
+                    # data)
                     if save_model == 'best':
 
-                        if min(valid_losses) == _val_loss:
-                            self.save_model(filename='t{}_y{}_b{}_e{}_lr{}_{}.pth.tar'.format(
-                                self.task, self.target, str(self.batch_size), str(nepoch), str(self.lr), str(epoch)))
+                        if min(valid_losses) == _val_loss: # noqa
+                            self.save_model(
+                                filename=f't{self.task}_y{self.target}_b{str(self.batch_size)}_e{str(nepoch)}_lr{str(self.lr)}_{str(epoch)}.pth.tar')
                 else:
-                    # if no validation set, saves the best performing model on the traing set
+                    # if no validation set, saves the best performing model on
+                    # the traing set
                     if save_model == 'best':
-                        if min(train_losses) == _loss:
-                            _log.warning("""The training set is used both for learning and model selection.
+                        if min(train_losses) == _loss: # noqa
```

Comment:
same as above, `_loss` is not defined. Should it be an input for `train` method?

Reply:
Please replace it by the 'loss_' variable.